scrapy_autounit/cassette.py

```diff
@@ -0,0 +1,80 @@
+import pickle
+import zlib
+
+from scrapy.crawler import Crawler
+from scrapy.utils.conf import build_component_list
+from scrapy.utils.project import get_project_settings
+
+from .utils import get_spider_class
+
+
+class Cassette:
+    """
+    Helper class to store request, response and output data.
+    """
+    FIXTURE_VERSION = 2
+
+    def __init__(self, spider=None):
+        if spider:
+            self.spider_name = spider.name
+            self.middlewares = self._get_middlewares(spider.settings)
+            self.included_settings = self._get_included_settings(spider.settings)
+        self.python_version = sys.version_info.major
+
+        # Set by Recorder.new_cassette
+        self.request = None
+        self.response = None
+        self.init_attrs = None
+        self.input_attrs = None
+
+        # Set by Recorder.record
+        self.filename = None
+        self.output_data = None
+        self.output_attrs = None
```

Comment:
What do you think about accepting the `request`, `response`, etc params in `__init__`, all `None` by default, instead of assigning them as attributes in `Recorder`'s methods?

Reply:
Sure, that works too 👍

Reply:
Fixed in https://github.com/scrapinghub/scrapy-autounit/commit/18334a37aad5b98671b414891268bddda4113de7